server/src/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -2378,6 +2378,23 @@ public boolean maintenanceFailed(final long hostId) {
     }
 
     @Override
+    public List<HostVO> listAllUpHosts(Type type, Long clusterId, Long podId, long dcId) {
```

Comment:
Hi @SudharmaJain

could you also write an unit test for this new method you created since it brings new functionalities to the code?